EKF/ekf_helper.cpp

```diff
@@ -1743,8 +1745,13 @@ bool Ekf::resetYawToEKFGSF()
 
```

Comment:
We should only set this flag `_control_status.flags.mag_fault = true;` if the reason for the reset is due to an "Emergency yaw reset", otherwise we always get the `mag_fault` flag when flying without magnetometer (this produces the PX4 error `WARN  [commander] Stopping compass use! Check calibration on landing`).

A quick fix is to place `_control_status.flags.mag_fault = true;` a few lines below, with the `ECL_INFO_TIMESTAMPED("Emergency yaw reset - magnetometer use stopped");`. 
IMO, this function `resetYawToEKFGSF` does too many things (i.e.: reset yaw + reset velocity + reset position + set mag fault flag + gives an info message depending on the type of mag fusion) but we can still clean it later, I'm fine with the minimal fix.

Reply:
Will fix